lib/vehicle_factory.rb

```diff
@@ -0,0 +1,45 @@
+class VehicleFactory
+
+    def initialize
+        @created_vehicles = []
+        @vehicle_data = {}
+    end
+
+    def create_vehicles(registrations)
+       registrations.each do |vehicle_hash|
+        @vehicle_data[:engine] = :ev
```

Comment:
Same thing here, don't need the vehicle data as an ivar. Otherwise, great job with this method!